kube-apiserver/tls-config.md

```diff
@@ -0,0 +1,366 @@
+
+The idea is to create a set of standard profiles: old, intermediate, modern, and custom that can be used.  Old, intermediate, 
+and modern are intent based.  We will update them as we see fit to stay up to date with standards as time goes on.  Custom
+allows a user to specify exactly which options they want.
+
+## Motivation
+
+We want guardrails where possible to avoid letting a customer make choices that prevent future upgrades.  Lots of flexibility
+is very easy to produce once, but very hard to maintain in an auto-updating world.  We will use profiles to constrain that choice.
+
+### Goals
+
+1. Easily configure a uniform level of TLSConfig (ciphers, tls versions, and DH key size) across kube-apiserver, oauth-server,
+ and all ingress controllers. These are the only known exposed services today.
+2. Allow overriding on a per-router basis.  Workloads and infrastructure may have different requirements and we need to 
+ support that use-case.
+
+### Non-Goals
+
+1. Make it easy to implement unusual configurations.
+2. Make decisions about future restrictions like tls curves.
+
+## Proposal
+
+See https://github.com/openshift/api/pull/432 for the details of the API to be included, but at a high level it looks like this
+
+```go
+// TLSSecurityProfile defines the schema for a TLS security profile. This object
+// is used by operators to apply TLS security settings to operands.
+// +union
+type TLSSecurityProfile struct {
+	// type is one of Old, Intermediate, Modern or Custom. Custom provides
+	// the ability to specify individual TLS security profile parameters.
+	// Old, Intermediate and Modern are TLS security profiles based on:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Recommended_configurations
+	//
+	// The profiles are intent based, so they may change over time as new ciphers are developed and existing ciphers
+	// are found to be insecure.  Depending on precisely which ciphers are available to a process, the list may be
+	// reduced.
+	//
+	// +unionDiscriminator
+	// +optional
+	Type TLSProfileType `json:"type"`
+	// old is a TLS security profile based on:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Old_backward_compatibility
+	//
+	// +optional
+	// +nullable
+	Old *OldTLSProfile `json:"old,omitempty"`
+	// intermediate is a TLS security profile based on:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	//
+	// +optional
+	// +nullable
+	Intermediate *IntermediateTLSProfile `json:"intermediate,omitempty"`
+	// modern is a TLS security profile based on:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Modern_compatibility
+	//
+	// +optional
+	// +nullable
+	Modern *ModernTLSProfile `json:"modern,omitempty"`
+	// custom is a user-defined TLS security profile. Be extremely careful using a custom
+	// profile as invalid configurations can be catastrophic. An example custom profile
+	// looks like this:
```

Comment:
no example is provided, so remove `An example custom profile
	// looks like this:`.